clients/scheduler/interval.go

```diff
@@ -49,84 +46,100 @@ type IntervalClient interface {
+	urlClient interfaces.URLClient
 }
 
 // NewIntervalClient creates an instance of IntervalClient
 func NewIntervalClient(params types.EndpointParams, m interfaces.Endpointer) IntervalClient {
-	s := intervalRestClient{endpoint: m}
-	s.init(params)
-	return &s
+	return &intervalRestClient{urlClient: urlclient.New(params, m)}
 }
 
-func (s *intervalRestClient) init(params types.EndpointParams) {
-	if params.UseRegistry {
-		go func(ch chan string) {
-			for {
-				select {
-				case url := <-ch:
-					s.url = url
-				}
-			}
-		}(s.endpoint.Monitor(params))
-	} else {
-		s.url = params.Url
+func (ic *intervalRestClient) Add(interval *models.Interval, ctx context.Context) (string, error) {
+	url, err := ic.urlClient.Prefix()
+	if err != nil {
+		return "", err
 	}
-}
 
-func (s *intervalRestClient) Add(interval *models.Interval, ctx context.Context) (string, error) {
-	return clients.PostJsonRequest(s.url, interval, ctx)
+	return clients.PostJsonRequest(url, interval, ctx)
 }
 
-func (s *intervalRestClient) Delete(id string, ctx context.Context) error {
-	return clients.DeleteRequest(s.url+"/id/"+id, ctx)
-}
+func (ic *intervalRestClient) Delete(id string, ctx context.Context) error {
+	urlPrefix, err := ic.urlClient.Prefix()
+	if err != nil {
+		return err
+	}
 
-func (s *intervalRestClient) DeleteByName(name string, ctx context.Context) error {
-	return clients.DeleteRequest(s.url+"/name/"+url.QueryEscape(name), ctx)
+	return clients.DeleteRequest(urlPrefix+"/id/"+id, ctx)
 }
 
-func (s *intervalRestClient) Interval(id string, ctx context.Context) (models.Interval, error) {
-	return s.requestInterval(s.url+"/"+id, ctx)
+func (ic *intervalRestClient) DeleteByName(name string, ctx context.Context) error {
+	urlPrefix, err := ic.urlClient.Prefix()
+	if err != nil {
+		return err
+	}
+
+	return clients.DeleteRequest(urlPrefix+"/name/"+url.QueryEscape(name), ctx)
 }
 
-func (s *intervalRestClient) IntervalForName(name string, ctx context.Context) (models.Interval, error) {
-	return s.requestInterval(s.url+"/name/"+url.QueryEscape(name), ctx)
+func (ic *intervalRestClient) Interval(id string, ctx context.Context) (models.Interval, error) {
+	return ic.requestInterval("/"+id, ctx)
 }
 
-func (s *intervalRestClient) Intervals(ctx context.Context) ([]models.Interval, error) {
-	return s.requestIntervalSlice(s.url, ctx)
+func (ic *intervalRestClient) IntervalForName(name string, ctx context.Context) (models.Interval, error) {
+	return ic.requestInterval("/name/"+url.QueryEscape(name), ctx)
 }
 
-func (s *intervalRestClient) Update(interval models.Interval, ctx context.Context) error {
-	return clients.UpdateRequest(s.url, interval, ctx)
+func (ic *intervalRestClient) Intervals(ctx context.Context) ([]models.Interval, error) {
+	return ic.requestIntervalSlice("", ctx)
 }
 
-//
-// Helper functions
-//
+func (ic *intervalRestClient) Update(interval models.Interval, ctx context.Context) error {
+	url, err := ic.urlClient.Prefix()
```

Comment:
Variable naming in prior PRs (and elsewhere in this same file) is `urlPrefix`; rename `url` here for consistency.  Applies to any/all calls to Prefix().

Reply:
The distinction I made was that when there's no suffix added I just named it "url"

Reply:
I see.  That wasn't immediately apparent to me.  Still prefer consistent naming of variable returned by Prefix(); `urlPrefix` makes the most sense to me.